coilsnake/modules/eb/CastModule.py

```diff
@@ -0,0 +1,283 @@
+import logging
+
+from coilsnake.exceptions.common.exceptions import InvalidArgumentError
+from coilsnake.model.common.ips import IpsPatch
+from coilsnake.model.eb.graphics import EbCastMiscGraphic, EbCastNameGraphic
+from coilsnake.model.eb.palettes import EbPalette
+from coilsnake.model.eb.table import EbStandardNullTerminatedTextTableEntry
+from coilsnake.modules.common.PatchModule import get_ips_filename
+from coilsnake.modules.eb.EbModule import EbModule
+from coilsnake.util.common.image import open_image, open_indexed_image
+from coilsnake.util.common.yml import yml_dump, yml_load
+from coilsnake.util.eb.pointer import from_snes_address, read_asm_pointer, to_snes_address, write_asm_pointer
+
+log = logging.getLogger(__name__)
+
+
+CAST_DYNAMIC_NAMES_FILE_NAME = 'Cast/dynamic_names'
+CAST_FORMATTING_FILE_NAME = 'Cast/graphics_entries'
+
+CAST_FORMATTING_ENTRY_SIZE = 3
+CAST_FORMATTING_OFFSET = 0x212efa + CAST_FORMATTING_ENTRY_SIZE # skip the first entry
+CAST_FORMATTING_TABLE_SIZE = 0x90 - CAST_FORMATTING_ENTRY_SIZE # first entry was skipped
+CAST_FORMATTING_OFFSET_END = CAST_FORMATTING_OFFSET + CAST_FORMATTING_TABLE_SIZE
+
+CAST_GRAPHICS_PALETTE_OFFSET = 0x21d815
+CAST_MISC_GRAPHICS_ASM_POINTER_OFFSET = 0x4e42e
+CAST_NAME_GRAPHICS_ASM_POINTER_OFFSET = 0x4e446
+
+CAST_GRAPHICS_BLOCK_BEGIN = 0x21d815
+CAST_GRAPHICS_BLOCK_END = 0x21e4e7
+
+DYNAMIC_CAST_NAME_MODES = ['none', 'prefix', 'suffix']
+
+class EbDynamicCastName(object):
+    def __init__(self, key, asm_pointer_offset, custom_asm_offset):
+        self.key = key
+        self.asm_pointer_offset = asm_pointer_offset
+        self.table_entry = EbStandardNullTerminatedTextTableEntry.create(12)
+        self.text = ''
+        self.mode = 'none'
+        self.patch_name_prefix = key.replace('\'', '').replace(' ', '_').lower()
```

Comment:
I'd recommend passing the patch name prefix as its own parameter rather than trying to figure it out from the key